expression/builtin_time.go

```diff
@@ -7252,11 +7189,19 @@ func calAppropriateTime(minTime, maxTime, minSafeTime time.Time) time.Time {
 	return minSafeTime
 }
 
-func getFspByIntArg(ctx sessionctx.Context, exp Expression) (int, error) {
-	_, ok := exp.(*Constant)
+// getFspByIntArg is used by some time functions to get the result fsp. If len(expr) == 0, then the fsp is not explicit set, use 0 as default.
+func getFspByIntArg(ctx sessionctx.Context, exps []Expression) (int, error) {
+	if len(exps) == 0 {
+		return 0, nil
+	}
+	if len(exps) != 1 {
+		return 0, errors.Errorf("Should not happen, the num of argument should be 1, but got %d", len(exps))
+	}
+	_, ok := exps[0].(*Constant)
 	if ok {
-		fsp, _, err := exp.EvalInt(ctx, chunk.Row{})
-		if err != nil {
+		fsp, isNuLL, err := exps[0].EvalInt(ctx, chunk.Row{})
+		if err != nil || isNuLL {
+			// If isNULL, it may be a bug of parser. Return 0 to be compatible with old version.
```

Comment:
Why do you change this? I think this behavior is exactly the same as before.

Reply:
Explicitly handle the `isNull=NULL`. Otherwise, it seems that we have a bug here.